cobigen/cobigen-core-parent/cobigen-core-systemtest/src/test/java/com/devonfw/cobigen/systemtest/GenerationTest.java

```diff
@@ -90,21 +90,25 @@ public void testGenerationWithExternalIncrements() throws Exception {
         List<IncrementTo> increments = cobigen.getMatchingIncrements(input);
         List<String> triggersIds = cobigen.getMatchingTriggerIds(input);
 
-        // assert
-        IncrementTo externalIncrement = null;
         // we try to get an increment containing external increments
         for (IncrementTo inc : increments) {
             if (inc.getId().equals("3")) {
-                externalIncrement = inc;
+                // We expect increment 3 to have an external increment 0 containing one template
+                assertThat(inc).isNotNull();
+                assertThat(inc.getDependentIncrements().get(0).getTemplates().size()).isEqualTo(1);
+            }
+            if (inc.getId().equals("4") || inc.getId().equals("5")) {
+                // We expect increment 4 or 5 to have an external increment 0 containing 4 templates
+                assertThat(inc).isNotNull();
+                assertThat(inc.getDependentIncrements().get(0).getTemplates().size()).isEqualTo(4);
             }
         }
```

Comment:
Be careful, you never assert that templates 3,4,5 are even existing in the list

Reply:
even better, in such a case you could use nice assertJ syntax for that like:

```java
assertThat(inc)
  .extracting(e -> e.getId(), e -> e.getDependentIncrements().get(0).getTemplates().size())
  .containsExactlyInAnyOrder(tuple("3", 1), tuple("4", 4), tuple("5", 4));
```

This will automatically check the existence of exactly three elements having exactly the properties you mentioned here.
There are alos other functions like containsExactly or contains depending on what you want to test.